src/Bridge/Symfony/Bundle/Action/RedocAction.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\Action;
+
+use ApiPlatform\Core\Documentation\Documentation;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Displays the documentation with ReDoc.
+ *
+ * @author Grégoire Hébert <gregoire@les-tilleuls.coop>
+ */
+final class RedocAction
+{
+    private $twig;
+    private $urlGenerator;
+    private $title;
+
+    public function __construct(\Twig_Environment $twig, UrlGeneratorInterface $urlGenerator, string $title = '')
+    {
+        $this->twig = $twig;
+        $this->urlGenerator = $urlGenerator;
+        $this->title = $title;
+    }
+
+    public function __invoke()
```

Comment:
Cannot we merge this with the existing listener for Swagger UI and just use a `if` in the Twig template according to the default and/or the query parameter?